File: Samples.Objc/Samples.ObjC/API/Responses/CreateOrderResponse.m
//
//  CreateOrderResponse.m
//  PayPalCheckout-Samples-iOS-Objc
//
//  Created by Haider Khan on 5/19/20.
//  Copyright © 2020 PayPal. All rights reserved.
//

#import "CreateOrderResponse.h"

@implementation CreateOrderResponse

- (id)initWithData:(NSData *)data {
  self = [super init];
  if (self) {
    NSError *error;
    NSDictionary *dictionaryFromData = [NSJSONSerialization
                                        JSONObjectWithData:data
                                        options:NSJSONReadingAllowFragments
                                        error:&error];
    if (error == nil) {
      if ([dictionaryFromData valueForKey:@"id"]) {
        NSString *orderId = [dictionaryFromData valueForKey:@"id"];
        self.orderId = orderId;
      }
    }
    else {
      self.orderId = @"";
    }
  }
  return self;
}

@end

File: Samples.Swift/Samples.Swift/API/Responses/CreateOrderResponse.swift
//
//  CreateOrderResponse.swift
//  paypalcheckout-iOS-swift-sample
//
//  Created by Jax DesMarais-Leder on 3/16/21.
//

import Foundation

class CreateOrderResponse: NSObject {
  let orderId: String

  init(data: Data) {
    let dictionaryFromData = try? JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: Any]

    let orderId = dictionaryFromData?["id"] as? String
    self.orderId = orderId ?? ""
  }
}
